frontend/src/test/scala/bloop/ConsoleSpec.scala

```diff
@@ -0,0 +1,55 @@
+package bloop
+
+import bloop.testing.BaseSuite
+import bloop.util.TestUtil
+import bloop.logging.RecordingLogger
+import bloop.util.TestProject
+import bloop.cli.ExitStatus
+import coursier.CoursierPaths
+import coursier.Cache
+import scala.util.Properties
+import java.nio.file.Paths
+
+object ConsoleSpec extends BaseSuite {
+  test("console works") {
+    TestUtil.withinWorkspace { workspace =>
+      object Sources {
+        val `A.scala` =
+          """/A.scala
+            |class A
+          """.stripMargin
+        val `B.scala` =
+          """/B.scala
+            |class B extends A
+          """.stripMargin
+      }
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val `A` = TestProject(workspace, "a", List(Sources.`A.scala`))
+      val `B` = TestProject(workspace, "b", List(Sources.`B.scala`), List(`A`))
+      val projects = List(`A`, `B`)
+      val state = loadState(workspace, projects, logger)
+      val compiledState = state.console(`B`)
+      assert(compiledState.status == ExitStatus.Ok)
+      assertValidCompilationState(compiledState, projects)
+
+      val cache = Paths.get(Properties.userHome, ".cache", "coursier", "v1")
```

Comment:
I believe this can break if the artifacts aren't read from cache. How should this be handled?

Reply:
The artifacts should always be in cache because coursier and sbt will first download them to the cache and then use the cached path, however I've rewritten this in my branch to generate the full classpath programmatically to be platform-independent (tests run on Windows and cache paths are not the same in macOS and Linux)